index.js

```diff
@@ -0,0 +1,75 @@
+document.addEventListener('DOMContentLoaded', function () {
+    const burgerBtn = document.querySelector('.navbar__burger-btn');
+    const navbarList = document.querySelector('.navbar__list');
+    const navbarItems = document.querySelectorAll('.navbar__item--has-submenu');
+    const submenuItems = document.querySelectorAll('.navbar__submenu-item--has-submenu');
+    const quizItems = document.querySelectorAll('.navbar__submenu-item--has-submenu-lowest');
+
+    burgerBtn.addEventListener('click', () => {
+      navbarList.classList.toggle('active');
+    });
+
+    const closeNavbarItems = () => {
+      navbarItems.forEach(item => {
+        item.classList.remove('active');
+      });
+    }
+
+    const closeSubmenuItems = () => {
+      submenuItems.forEach(item => {
+        item.classList.remove('active');
+      })
+    }
+
+    const closeQuizItems = () => {
+      quizItems.forEach(item => {
+        item.classList.remove('active');
+      })
+    }
+  
+    navbarItems.forEach(item => {
+      item.addEventListener('click', function (e) {
+        e.stopPropagation();
+        const isActive = this.classList.contains('active');
+        closeNavbarItems();
+        if(!isActive) {
+          this.classList.add('active');
+        }
+      });
+    });
+
+    submenuItems.forEach(item => {
+        item.addEventListener('click', function (e) {
+          e.stopPropagation();
+          const isActive = this.classList.contains('active');
+          closeSubmenuItems();
+
+          if(!isActive) {
+            this.classList.add('active');
+          }
+          console.log('Clicked navitem')
+        });
+    });
+
+    quizItems.forEach(item => {
+      item.addEventListener('click', function (e) {
+        e.stopPropagation();
+        const isActive = this.classList.contains('active');
+        closeQuizItems();
+
+        if(!isActive) {
+          this.classList.add('active');
+        }
+      })
+    })
+  
+    document.addEventListener('click', function (e) {
+      if(!navbarList.contains(e.target)) {
+        closeNavbarItems();
+      }
+
+      if(!Array.from(submenuItems).some(item => item.contains(e.target))) {
+        closeSubmenuItems();
+      }
+    })
+  });
```

Comment:
this file looks ok like for first iteration, but let's have a quick exercise - how could you make it more reusable? Do you have any idea how to use more generic methods here? Please state your answer in comment without changing the code.

Reply:
I see your point, there's quite lot code repetition. I guess I could create one function for closing the items and one  function for adding click listener, that will dynamically accept the element as an argument at which operations will be performed?

Reply:
Yep:

- **Reduce repetition**: Combine the `closeNavbarItems`, `closeSubmenuItems`, and `closeQuizItems` functions into a single reusable function that takes a selector as an argument to close items.
- **Handle events dynamically**: Instead of duplicating the event listener logic for each type of item, create a generic function that handles item clicks and can be reused across different selectors.
- **Consider event delegation**: Delegate click events to a common parent (like `navbarList`) to handle clicks more efficiently and simplify the logic.

However, **be cautious with over-generalizing** your code. Making it too abstract can result in code that's harder to maintain than if you handle each case individually. Keep a balance between reusability and clarity